File: app/src/main/java/org/d3ifcool/jagosholat/views/activities/MainActivity.java
package org.d3ifcool.jagosholat.views.activities;

/*
 * Created by Faisal Amir
 * FrogoBox Inc License
 * =========================================
 * JagoSholat
 * Copyright (C) 25/04/2018.
 * All rights reserved
 * -----------------------------------------
 * Name     : Muhammad Faisal Amir
 * E-mail   : faisalamircs@gmail.com
 * Line     : bullbee117
 * Phone    : 081357108568
 * Majors   : D3 Teknik Informatika 2016
 * Campus   : Telkom University
 * -----------------------------------------
 * id.amirisback.frogobox
 */

import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import org.d3ifcool.jagosholat.R;
import org.d3ifcool.jagosholat.views.fragments.CatatanFragment;
import org.d3ifcool.jagosholat.views.fragments.JadwalFragment;
import org.d3ifcool.jagosholat.views.fragments.KiblatFragment;
import org.d3ifcool.jagosholat.views.fragments.StatistikFragment;
import org.d3ifcool.jagosholat.views.fragments.TataCaraFragment;
import org.d3ifcool.jagosholat.views.interfaces.ClickHandler;

import static org.d3ifcool.jagosholat.models.constants.VarConstants.Constants;

public class MainActivity extends AppCompatActivity implements ClickHandler {

    // ---------------------------------------------------------------------------------------------
    private boolean isTwoPane;
    private String mSelectedMenuTab;
    // ---------------------------------------------------------------------------------------------

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // -----------------------------------------------------------------------------------------
        isTwoPane = findViewById(R.id.detail_frame_layout) != null;
        Toolbar toolbar = findViewById(R.id.main_toolbar);
        setSupportActionBar(toolbar);
        // -----------------------------------------------------------------------------------------
        if (isTwoPane) {
            mSelectedMenuTab = savedInstanceState!=null ?
                    savedInstanceState.getString(Constants.STRING_EXTRA_FRAGMENT):Constants.TAB_CATATAN;
            menuTabClick(mSelectedMenuTab);
        }
        // -----------------------------------------------------------------------------------------

    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        if (isTwoPane) outState.putString(Constants.STRING_EXTRA_FRAGMENT, mSelectedMenuTab);
        super.onSaveInstanceState(outState);
    }

    @Override
    public void menuTabClick(String menuTab) {
        if (isTwoPane) {
            mSelectedMenuTab = Constants.STRING_EXTRA_FRAGMENT;
            Fragment mFragment;
            switch (menuTab) {
                case Constants.TAB_CATATAN:
                    mFragment = new CatatanFragment();
                    break;
                case Constants.TAB_JADWAL:
                    mFragment = new JadwalFragment();
                    break;
                case Constants.TAB_STATISTIK:
                    mFragment = new StatistikFragment();
                    break;
                case Constants.TAB_KIBLAT:
                    mFragment = new KiblatFragment();
                    break;
                case Constants.TAB_TATACARA:
                    mFragment = new TataCaraFragment();
                    break;
                default:
                    throw new IllegalArgumentException();
            }
            FragmentManager mFragmentManager = getSupportFragmentManager();
            mFragmentManager.beginTransaction().replace(R.id.detail_frame_layout, mFragment).commit();
        } else {
            // Create a new intent to open the {@link NumbersFragment}
            Intent mIntent = new Intent(this, DetailActivity.class);
            mIntent.putExtra(Constants.STRING_EXTRA_FRAGMENT, menuTab);
            // Start the new activity
            startActivity(mIntent);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        // Membuat Tombol Menu di Ujung Kanan Aplikasi
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Melihat Info Apps dengan cara Toolbar
        switch (item.getItemId()) {
            case R.id.info_apps:
                Intent i = new Intent(MainActivity.this, TentangKamiActivity.class);
                startActivity(i);
                return true;
        }
        //noinspection SimplifiableIfStatemen
        return super.onOptionsItemSelected(item);
    }

    public int getActionBarSize() {
        TypedArray styledAttributes = getTheme().obtainStyledAttributes(new int[] { android.R.attr.actionBarSize });
        int actionBarSize = (int) styledAttributes.getDimension(0, 0);
        styledAttributes.recycle();
        return actionBarSize;
    }

}

File: app/src/main/java/com/frogobox/praybox/ui/activitiy/MainActivity.kt
package com.frogobox.praybox.ui.activitiy

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.frogobox.praybox.R
import com.frogobox.praybox.base.view.ui.BaseActivity
import com.frogobox.praybox.ui.fragment.CatatanFragment
import com.frogobox.praybox.ui.fragment.JadwalFragment
import com.frogobox.praybox.ui.fragment.StatistikFragment
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.toolbar_main.*

/*
 * Created by Faisal Amir
 * FrogoBox Inc License
 * =========================================
 * JagoSholat
 * Copyright (C) 25/04/2018.
 * All rights reserved
 * -----------------------------------------
 * Name     : Muhammad Faisal Amir
 * E-mail   : faisalamircs@gmail.com
 * Line     : bullbee117
 * Phone    : 081357108568
 * Majors   : D3 Teknik Informatika 2016
 * Campus   : Telkom University
 * -----------------------------------------
 * id.amirisback.frogobox
 */

class MainActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupToolbar()
        setupBottomNav(R.id.framelayout_main_container)
        setupFragment(savedInstanceState)
    }

    private fun setupFragment(savedInstanceState: Bundle?) {
        if (savedInstanceState == null) {
            bottom_nav_main_menu.selectedItemId = R.id.bottom_menu_catatan
        }
    }

    private fun setupToolbar() {
        setSupportActionBar(toolbar_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_toolbar_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.toolbar_menu_about -> {
                baseStartActivity<TentangKamiActivity>()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun setupBottomNav(frameLayout: Int) {
        bottom_nav_main_menu.clearAnimation()
        bottom_nav_main_menu.itemIconTintList = null
        bottom_nav_main_menu.setOnNavigationItemSelectedListener {

            when (it.itemId) {
                R.id.bottom_menu_catatan -> {
                    setupCustomTitleToolbar(R.string.menu_catatan)
                    setupChildFragment(frameLayout, CatatanFragment())
                }

                R.id.bottom_menu_statistik -> {
                    setupCustomTitleToolbar(R.string.menu_statistik)
                    setupChildFragment(frameLayout, StatistikFragment())
                }

                R.id.bottom_menu_jadwal -> {
                    setupCustomTitleToolbar(R.string.menu_jadwal)
                    setupChildFragment(frameLayout, JadwalFragment())
                }

            }

            true
        }

    }
}